x/ibc/04-channel/keeper/keeper.go

```diff
@@ -55,15 +55,26 @@ func (k Keeper) GetChannel(ctx sdk.Context, portID, channelID string) (types.Cha
 		return types.Channel{}, false
 	}
 
-	var channel types.Channel
-	k.cdc.MustUnmarshalBinaryBare(bz, &channel)
-	return channel, true
+	channelI, err := k.cdc.UnmarshalChannel(bz)
+	if err != nil {
+		panic(err)
+	}
+
+	channel, ok := channelI.(*types.Channel)
```

Comment:
Same question I have here that I had for Connection. Why not decode directly into a Channel? Are there multiple implementations?